openhtf/exe/test_state.py

```diff
@@ -95,33 +82,36 @@ def _asdict(self):
         'status': self._state.name,
         'record': self.record,
         'phase_data': self.phase_data,
-        'running_phase': self.running_phase,
+        'running_phase_record': self.running_phase_record,
         'pending_phases': [(phase.func.__name__, phase.func.__doc__)
                            for phase in self.pending_phases]}
 
-  def SetStateFromPhaseResult(self, phase_result):
-    """Set our internal state based on the given phase result.
+  def SetStateFromPhaseOutcome(self, phase_outcome):
+    """Set our internal state based on the given phase outcome.
 
     Args:
-      phase_result: An instance of phasemanager.TestPhaseResult
+      phase_outcome: An instance of phase_executor.PhaseOutcome
 
-    Returns: True if the test has finished.
+    Returns: True if the test has finished prematurely (failed).
     """
-    if phase_result.raised_exception:
-      self._state = self.State.ERROR
-      code = str(type(phase_result.phase_result).__name__)
-      details = str(phase_result.phase_result).decode('utf8', 'replace')
-      self.record.AddOutcomeDetails(self._state, code, details)
-    else:
-      if phase_result.phase_result not in self._PHASE_RESULT_TO_STATE:
-        raise InvalidPhaseResultError(
-            'Phase result is invalid.', phase_result.phase_result)
-      self._state = self._PHASE_RESULT_TO_STATE[phase_result.phase_result]
-
-    return self._state in self._FINISHED_STATES
+    # Handle a few cases where the test is ending prematurely.
+    if phase_outcome.raised_exception:
+      self.record.outcome = 'ERROR'
```

Comment:
Okay, after looking into this more, we were doing a poor job with enums in a couple places, both measurement outcomes and test outcome.  I've converted both to use Enum, and I think it looks a whole lot cleaner now, thanks for the suggestion :)  I've also updated the output mechanisms to stringify enums to their name, so the end result is we get the same string output, but internally they are proper Enum instances.